pkg/apis/garden/validation/validation.go

```diff
@@ -312,41 +312,41 @@ func validateAlicloudMachineTypeConstraints(machineTypes []garden.AlicloudMachin
 	return allErrs
 }
 
-func validateMachineImageNames(names []string, fldPath *field.Path) field.ErrorList {
+func validateMachineImages(machineImages []garden.MachineImage, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	if len(names) == 0 {
-		allErrs = append(allErrs, field.Required(fldPath, "must provide at least one machine image"))
-		return allErrs
+	if len(machineImages) == 0 {
+ 		allErrs = append(allErrs, field.Required(fldPath, "must provide at least one machine image"))
 	}
 
-	imageNames := map[string]bool{}
-	for i, name := range names {
-		idxPath := fldPath.Index(i)
-
-		if imageNames[name] {
-			allErrs = append(allErrs, field.Duplicate(idxPath, name))
-		}
-		imageNames[name] = true
+	_, latestMachineImage, err := helper.DetermineLatestMachineImage(machineImages)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, latestMachineImage, "failed to determine latest machine image from cloud profile"))
+	}
+	if  latestMachineImage.ExpirationDate != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("expirationDate"), latestMachineImage.ExpirationDate, "expiration date of latest machine image must not be set"))
 	}
```

Comment:
Also, please validate that the specified expiration date is in the future and did not already expire.